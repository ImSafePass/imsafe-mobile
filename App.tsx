import React from "react";
import { StyleSheet, Text, View } from "react-native";

import AuthenticatedHeader from "./shared/components/AuthenticatedHeader";
import { padding } from "./shared/utils/style/dims";
import { white } from "./shared/utils/style/colors";

const App = () => (
  <View style={styles.container}>
    <AuthenticatedHeader />
    <View style={styles.main}>
      <Text style={styles.header}>Welcome to ImSafe!</Text>
    </View>
  </View>
);

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: white,
    alignItems: "center",
    justifyContent: "flex-start",
  },
  header: {
    fontSize: 40,
  },
  main: {
    flexDirection: "column",
    justifyContent: "flex-start",
    padding: padding,
  },
});

export default App;
