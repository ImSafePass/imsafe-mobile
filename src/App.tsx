import React from "react";
import { StyleSheet, View, Platform } from "react-native";

import { H1, H4 } from "./shared/components/Text";
import AuthenticatedHeader from "./shared/components/AuthenticatedHeader";
import { padding } from "./shared/utils/style/dims";
import { white } from "./shared/utils/style/colors";

const App = () => (
  <View style={styles.container}>
    <AuthenticatedHeader />
    <View style={styles.main}>
      <H1>Welcome to ImSafe!</H1>
      <H4>You're on {Platform.OS}</H4>
    </View>
  </View>
);

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: white,
    alignItems: "center",
    justifyContent: "flex-start",
  },
  main: {
    flexDirection: "column",
    justifyContent: "flex-start",
    padding: padding,
  },
});

export default App;
