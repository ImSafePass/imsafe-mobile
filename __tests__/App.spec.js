import "react-native";
import React from "react";
import { BaseApp } from "../src/App.tsx";

// Note: test renderer must be required after react-native.
import renderer from "react-test-renderer";

it("renders correctly", () => {
  const app = renderer.create(<BaseApp />);
  expect(app).toMatchSnapshot();
});
