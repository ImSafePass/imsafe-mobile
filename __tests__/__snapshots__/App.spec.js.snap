// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "flex-start",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#3341c0",
        "flexDirection": "row",
        "height": 80,
        "justifyContent": "space-between",
        "maxHeight": 80,
        "paddingHorizontal": 15,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flexDirection": "row",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "justifyContent": "center",
            },
            Object {
              "height": 40,
              "width": 30,
            },
          ]
        }
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight="100%"
          bbWidth="100%"
          fill="none"
          focusable={false}
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              Object {
                "flex": 0,
                "height": "100%",
                "width": "100%",
              },
            ]
          }
          vbHeight={49}
          vbWidth={36}
        >
          <RNSVGGroup
            fill={null}
            propList={
              Array [
                "fill",
              ]
            }
          >
            <RNSVGPath
              d="M18 0L0 6.86598V14.5988V15.2165V28.5047C0 35.9338 7.98081 41.3127 15.7874 46.5742C16.5199 47.0679 17.2508 47.5606 17.9735 48.0538C18.6962 47.5606 19.4271 47.0679 20.1596 46.5742C27.9662 41.3127 35.947 35.9338 35.947 28.5047V15.2165H36V6.86598L18 0Z"
              fill={4294244603}
              propList={
                Array [
                  "fill",
                ]
              }
            />
            <RNSVGPath
              d="M27.4152 18.088C26.7675 16.8929 22.707 12.8907 17.9999 18.685C13.0567 12.8907 9.23126 16.8929 8.58468 18.088C7.40735 20.298 8.11353 23.6437 9.76201 25.2567L18.0011 33.6205L26.2401 25.2567C27.8864 23.6437 28.5925 20.2991 27.4152 18.088H27.4152Z"
              fill={4281549248}
              propList={
                Array [
                  "fill",
                ]
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
      <Text
        style={
          Object {
            "color": "#fff",
            "fontFamily": "Heebo-Regular",
            "fontSize": 24,
            "marginLeft": 10,
          }
        }
      >
        ImSafe Pass
      </Text>
    </View>
    <View
      style={
        Object {
          "height": 40,
          "width": 40,
        }
      }
    >
      <Text />
    </View>
  </View>
  <View
    style={
      Object {
        "flexDirection": "column",
        "justifyContent": "flex-start",
        "padding": 15,
      }
    }
  >
    <Text
      style={
        Object {
          "fontFamily": "Heebo-ExtraBold",
          "fontSize": 40,
        }
      }
    >
      Welcome to ImSafe!
    </Text>
    <Text
      style={
        Object {
          "fontFamily": "Heebo-Regular",
          "fontSize": 24,
        }
      }
    >
      You're on 
      ios
    </Text>
  </View>
</View>
`;
